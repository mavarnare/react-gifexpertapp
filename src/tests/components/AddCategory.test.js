import React from 'react';
import '@testing-library/jest-dom'; //---> Para que aparezcan las ayudas de la librearía jest

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    const preventDefault = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    });

    test('debe renderizar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });
        
        expect(wrapper.find('input').prop('value')).toBe(value);
    })

    test('no debe postear la información on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toHaveBeenCalled();
        expect(setCategories).not.toHaveBeenCalled();
    })

    test('debe llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Texto válido';
        wrapper.find('input').simulate('change', { target: { value } });
        expect(wrapper.find('input').prop('value')).toBe(value);
        wrapper.find('form').simulate('submit', { preventDefault });
        expect(wrapper.find('input').prop('value')).toBe('');
        expect(preventDefault).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
    })
    
})