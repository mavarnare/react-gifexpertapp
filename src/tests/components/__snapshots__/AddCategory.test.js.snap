// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe renderizar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
